catalog/core/catalog-core-ftp/src/main/java/ddf/catalog/core/ftp/ftplets/CustomFtplet.java

```diff
@@ -0,0 +1,368 @@
+import org.apache.ftpserver.ftplet.DefaultFtpReply;
+import org.apache.ftpserver.ftplet.DefaultFtplet;
+import org.apache.ftpserver.ftplet.FtpException;
+import org.apache.ftpserver.ftplet.FtpFile;
+import org.apache.ftpserver.ftplet.FtpReply;
+import org.apache.ftpserver.ftplet.FtpRequest;
+import org.apache.ftpserver.ftplet.FtpSession;
+import org.apache.ftpserver.ftplet.FtpletResult;
+import org.apache.ftpserver.impl.IODataConnectionFactory;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.codice.ddf.security.handler.api.UPAuthenticationToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.FileBackedOutputStream;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.data.impl.ContentItemImpl;
+import ddf.catalog.content.operation.CreateStorageRequest;
+import ddf.catalog.content.operation.impl.CreateStorageRequestImpl;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.mime.MimeTypeMapper;
+import ddf.mime.MimeTypeResolutionException;
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+
+public class CustomFtplet extends DefaultFtplet {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomFtplet.class);
+
+    private static final String UNSUPPORTED_METHOD_RESPONSE =
+            "Unsupported method, only PUT and MPUT are supported.";
+
+    private static final String SUBJECT = "subject";
+
+    private SecurityManager securityManager;
+
+    private CatalogFramework catalogFramework;
+
+    private MimeTypeMapper mimeTypeMapper;
+
+    public CustomFtplet(SecurityManager securityManager, CatalogFramework catalogFramework,
+            MimeTypeMapper mimeTypeMapper) {
+        this.securityManager = securityManager;
+        this.catalogFramework = catalogFramework;
+        this.mimeTypeMapper = mimeTypeMapper;
+    }
+
+    @Override
+    public FtpletResult onConnect(FtpSession session) throws FtpException, IOException {
+        return FtpletResult.DEFAULT;
+    }
+
+    @Override
+    public FtpletResult onDisconnect(FtpSession session) throws FtpException, IOException {
+        return FtpletResult.DEFAULT;
+    }
+
+    @Override
+    public FtpletResult onLogin(FtpSession session, FtpRequest request)
+            throws FtpException, IOException {
+        if (!StringUtils.isNotEmpty(request.getArgument())) {
+            session.write(new DefaultFtpReply(FtpReply.REPLY_451_REQUESTED_ACTION_ABORTED,
+                    "Password cannot be empty"));
+            return FtpletResult.DISCONNECT;
+        }
+
+        AuthenticationToken authenticationToken = new UPAuthenticationToken(session.getUser()
```

Comment:
Might want to add some TODOs in here for the security ticket.  We made need to check roles here and send a disconnect if they do not have CREATE permissions.  Also we may need to audit this request.

Paging @codice/security for more input.